crypto/fipsmodule/curve25519/curve25519.c

```diff
@@ -117,6 +117,7 @@ static void ed25519_keypair_pct(uint8_t public_key[ED25519_PUBLIC_KEY_LEN],
 
 void ED25519_keypair(uint8_t out_public_key[ED25519_PUBLIC_KEY_LEN],
   uint8_t out_private_key[ED25519_PRIVATE_KEY_LEN]) {
+  FIPS_service_indicator_lock_state();
```

Comment:
Because of the distance between the lock and unlock statements, it's helpful here to add a comment about which service(s) are prevented from updating the service indicator.